.github/workflows/cibuildwheel.yml

```
@@ -135,6 +135,24 @@ jobs:
           name: sdist
           path: dist/
 
+      - name: Build wheels for CPython 3.12
+        uses: pypa/cibuildwheel@39a63b5912f086dd459cf6fcb13dcdd3fe3bc24d # v2.15.0
+        with:
+          package-dir: dist/${{ needs.build_sdist.outputs.SDIST_NAME }}
+        env:
+          CIBW_BUILD: "cp312-*"
+          CIBW_ARCHS: ${{ matrix.cibw_archs }}
+          # Remove this once NumPy with Python 3.12 wheels is not pre-release.
+          CIBW_BEFORE_BUILD: >-
+            pip install certifi "pybind11>=2.6" "setuptools>=42" "setuptools_scm>=7" &&
```

Would you be willing to try adding contourpy's [build-requires](https://github.com/contourpy/contourpy/blob/c963ecc77ec3ab12b7259cf0fef8c1205479adf3/pyproject.toml#L4-L6)? They're not likely to put out a wheel until numpy puts out a final 1.26.0 release, which is not going to happen until 3.12 final.

```suggestion
            pip install certifi "setuptools>=42" "setuptools_scm>=7" "meson >= 1.2.0" "meson-python >= 0.13.1" "pybind11 >= 2.10.4"
```

---

Where are you getting this from ? CPython 3.12 .0 is scheduled for early October while numpy 1.26 's milestone is currently aimed at September 1st.

---

Possibly a faulty memory. Apologies.

---

no problem, just asking :)